lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
```

Comment:
What is the reasoning for needing a logging attribute here?

Couldn't we just get the log information from the config? (I believe we already do this in some config examples, maybe we should just continue that pattern and loose this logger param)